src/Cli/dotnet/commands/dotnet-workload/list/WorkloadListCommand.cs

```diff
@@ -40,23 +42,26 @@ internal class WorkloadListCommand : CommandBase
             var workloadManifestProvider = new SdkDirectoryWorkloadManifestProvider(dotnetPath, sdkVersion.ToString());
             var workloadResolver = WorkloadResolver.Create(workloadManifestProvider, dotnetPath, sdkVersion.ToString());
             _sdkFeatureBand = new SdkFeatureBand(sdkVersion);
-            _workloadRecordRepo = workloadRecordRepo ?? 
-                WorkloadInstallerFactory.GetWorkloadInstaller(_reporter, _sdkFeatureBand, workloadResolver, _verbosity).GetWorkloadInstallationRecordRepository();
+            _workloadRecordRepo = workloadRecordRepo ??
+                                  WorkloadInstallerFactory
+                                      .GetWorkloadInstaller(_reporter, _sdkFeatureBand, workloadResolver, _verbosity)
+                                      .GetWorkloadInstallationRecordRepository();
+            _sdkVersion = result.ValueForOption<string>(WorkloadUpdateCommandParser.SdkVersionOption);
         }
 
         public override int Execute()
         {
             var installedList = _workloadRecordRepo.GetInstalledWorkloads(_sdkFeatureBand);
             if (_machineReadableOption)
             {
-                var outputJson = new Dictionary<string, string[]>()
-                {
-                    ["installed"] = installedList.Select(id => id.ToString()).ToArray()
-                };
+                var updateAvailable = MockUpdateAvailable();
+                var listOutput = new ListOutput(Installed: installedList.Select(id => id.ToString()).ToArray(),
+                    UpdateAvailable: updateAvailable);
```

Comment:
So we expect to display updates by default when the machine readable option is present? Or is there a switch for displaying updates?

Reply:
Just one flag to display both. Since even for VSMac installer, if a workload has update available, it will be marked as "update available" instead.

Reply:
that reminds me, what happens if I run `dotnet workload install android` when the android workload is installed already? Will it get updated, get an error or do nothing

Reply:
It should get updated